sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -301,6 +883,85 @@ int main() {
 ...
 ----
 
+=== Diamond Dependency
+
+The following snippet of code shows how a SYCL `queue` can be put into a
+recording state, which allows a `command_graph` object to be populated by the
+command-groups submitted to the queue. Once the graph is complete, recording
+finishes on the queue to put it back into the default executing state. The
+graph is then finalized so that no more nodes can be added. Lastly, the graph is
+submitted as a whole for execution via
+`queue::submit(command_graph<graph_state::executable>)`.
+
+[source, c++]
+----
+  queue q{default_selector{}};
+
+  // New object representing graph of command-groups
+  ext::oneapi::experimental::command_graph graph;
+  {
+    buffer<T> bufferA{dataA.data(), range<1>{elements}};
+    buffer<T> bufferB{dataB.data(), range<1>{elements}};
+    buffer<T> bufferC{dataC.data(), range<1>{elements}};
+
+    // `q` will be put in the recording state where commands are recorded to
+    // `graph` rather than submitted for execution immediately.
+    q.begin_recording(graph);
+
+    // Record commands to `graph` with the following topology.
```

Comment:
Should we add the topology analogously to the dot product example?

Reply:
Could you clarify what change you'd like to see here? The samples are doing different things, so have different topologies. We could make both record & replay and explicit API have the same sample but using the two different mechanisms, although it seems like reductions are problematic in the implementation at the moment.

What I have done is add an ASCII diagram to the top of the dot product example like the one here, to illustrate the current different in topologies

Reply:
Going to resolve this thread, and we can do future sample updates in a later PR if desired.